.example/tests/integration/pack.sh

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+# This script is executed in this directory via `just pack-k8s` or `just pack-machine`.
+# Extra args are passed to this script, e.g. `just pack-k8s foo` -> $1 is 'foo'.
+# In CI, the `just pack-<substrate>` commands are invoked:
+#     - If this file exists and `just integration-<substrate>` would execute any tests
+#     - Before running integration tests
+#     - With no additional arguments
+#
+# Environment variables:
+# $CHARMLIBS_SUBSTRATE will have the value 'k8s' or 'machine' (set by pack-k8s or pack-machine)
+# In CI, $CHARMLIBS_TAG is set based on pyproject.toml:tool.charmlibs.integration.tags
+# For local testing, set $CHARMLIBS_TAG directly or use the tag variable. For example:
+# just tag=24.04 pack-k8s some extra args
+set -xueo pipefail
+
+TMP_DIR=".tmp"  # clean temporary directory where charms will be packed
+PACKED_DIR=".packed"  # where packed charms will be placed with name expected in conftest.py
+
+: copy charm files to temporary directory for packing, dereferencing symlinks
```

Comment:
I've never seen this before (shows how shallow my shell knowledge is). Why is it better than just using a `#` comment?